devops/scripts/benchmarks/main.py

```diff
@@ -363,6 +403,17 @@ def main(directory, additional_env_vars, compare_names, filter):
                 os.path.join(os.path.dirname(__file__), "html")
             )
         log.info(f"Generating HTML with benchmark results in {html_path}...")
+
+        # Reload history after saving to include the current results in HTML output
+        if not options.dry_run:
+            log.debug(
+                "Reloading history for HTML generation to include current results..."
+            )
+            history.load()
```

Comment:
This is a costly operation. I think that calling `history.save()` should add the latest run to the `history` object. Could you verify that this second load is really needed?